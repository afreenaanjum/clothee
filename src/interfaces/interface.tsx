import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {ReactNode} from 'react';
import {ViewStyle} from 'react-native';
import {RootStackParamList} from '../navigation/RouteStackParamList';
import {RoutesConstants} from '../utils/RouteEnums';

export interface IContainerProps {
  children: ReactNode;
  style?: ViewStyle;
}

export interface IProduct {
  'SERIAL NO': number;
  NAME: string;
  CATEGORY: string;
  'DESCRIPTION & COLOR': string;
  FABRIC: string;
  IMAGE: string;
  SIZE: string;
  PRICE: number;
  'PRODUCT ID': number;
  WEBSITE: string;
  'PRODUCT URL': string;
}

export interface IProps {
  navigation: NativeStackNavigationProp<
    RootStackParamList,
    RoutesConstants.HOME_SCREEN
  >;
}

// export interface IPdpProps {
//   navigation: NativeStackNavigationProp<
//     RootStackParamList,
//     RoutesConstants.PRODUCT_DETAILS_SCREEN
//   >;
// }

export interface IProductCardProps {
  product: IProduct;
}
