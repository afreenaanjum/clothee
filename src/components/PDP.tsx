import React from 'react';
import {StyleSheet, Text, Image, ActivityIndicator} from 'react-native';
import {IProductCardProps} from '../interfaces/interface';
import Container from './Container';

const PDP: React.FC<IProductCardProps> = ({product}: IProductCardProps) => {
  return (
    <Container style={styles.card}>
      <Container style={styles.imageContainer}>
        <Image
          source={{uri: product.IMAGE, cache: 'force-cache'}}
          onProgress={() => <ActivityIndicator />}
          resizeMode="contain"
          style={styles.h100}
        />
      </Container>
      <Container>
        <Text numberOfLines={2} ellipsizeMode="tail" style={styles.productName}>
          {product.NAME}
        </Text>
        <Text style={styles.productPrice}>Rs {product.PRICE}</Text>
        <Text style={styles.productDes}>{product['DESCRIPTION & COLOR']}</Text>
      </Container>
    </Container>
  );
};

const styles = StyleSheet.create({
  card: {
    backgroundColor: 'white',
    padding: 16,
    borderRadius: 5,
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
  },
  imageContainer: {aspectRatio: 1, alignSelf: 'center'},
  productDes: {
    color: 'grey',
  },
  productPrice: {
    fontWeight: '600',
  },
  h100: {
    height: '100%',
  },
  productName: {
    fontSize: 16,
    fontWeight: '500',
    color: '#FF1696',
    marginTop: 10,
  },
  w100: {minWidth: '100%'},
});

export default PDP;
