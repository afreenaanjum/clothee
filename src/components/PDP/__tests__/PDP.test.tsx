import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import PDP from '../PDP';
import {IProduct} from '../../../interfaces/interface';

describe('Center widget', () => {
  it('renders correctly', () => {
    let product: IProduct = {
      'SERIAL NO': 68,
      NAME: 'Baggit Men Brown Textured Card Holder',
      CATEGORY: 'accessories',
      'DESCRIPTION & COLOR':
        'Baggit Men Brown Textured Card Holder,  Baggit, Wallets, Wallets, Accessories, Accessories for Men, Buy Baggit Men Brown Textured Card Holder Online in India, Buy Wallets at best price',
      FABRIC: 'Synthetic',
      IMAGE:
        'https://assets.myntassets.com/h_1440,q_100,w_1080/v1/assets/images/6525475/2018/5/31/2fe78578-d217-44b0-9ada-d7cdd9f463cd1527762707750-Baggit-Men-Wallets-5631527762707516-1.jpg',
      SIZE: 'Height: 8.5 cm',
      PRICE: 540,
      'PRODUCT ID': 6525475,
      WEBSITE: 'Myntra',
      'PRODUCT URL':
        'https://www.myntra.com/wallets/baggit/baggit-men-brown-textured-card-holder/6525475/buy',
    };

    const tree = renderer.create(<PDP product={product} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  //   it('renders correctly when no children', () => {
  //     const tree = renderer.create(<PDP product={null} />).toJSON();
  //     expect(tree).toMatchSnapshot();
  //   });
});
