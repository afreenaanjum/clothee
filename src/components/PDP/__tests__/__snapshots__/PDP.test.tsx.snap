// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center widget renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "white",
        "borderRadius": 5,
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "padding": 16,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "alignSelf": "center",
          "aspectRatio": 1,
        },
      ]
    }
  >
    <Image
      onProgress={[Function]}
      resizeMode="contain"
      source={
        Object {
          "cache": "force-cache",
          "uri": "https://assets.myntassets.com/h_1440,q_100,w_1080/v1/assets/images/6525475/2018/5/31/2fe78578-d217-44b0-9ada-d7cdd9f463cd1527762707750-Baggit-Men-Wallets-5631527762707516-1.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Object {
          "color": "#FF1696",
          "fontSize": 16,
          "fontWeight": "500",
          "marginTop": 10,
        }
      }
    >
      Baggit Men Brown Textured Card Holder
    </Text>
    <Text
      style={
        Object {
          "fontWeight": "600",
        }
      }
    >
      Rs 
      540
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Baggit Men Brown Textured Card Holder,  Baggit, Wallets, Wallets, Accessories, Accessories for Men, Buy Baggit Men Brown Textured Card Holder Online in India, Buy Wallets at best price
    </Text>
  </View>
</View>
`;
